addons/website/models/website.py

```diff
@@ -1237,6 +1237,15 @@ def _enumerate_pages(self, query_string=None, force=False):
                 record['lastmod'] = page['write_date'].date()
             yield record
 
+    def get_website_page_ids(self):
+        domain = [('url', '!=', False)]
+        if self:
+            domain = AND([domain, self.website_domain()])
+        pages = self.env['website.page'].sudo().search(domain)
```

Comment:
Do you need that `sudo`? If you are only reading the website.page records and this is called in the context of someone being on the list view listing those pages... that someone necessarily have the correct reading rights no?

Reply:
Restricted editor don't have to right to read website.pages (no one has, apart from designer and +)
See https://github.com/odoo/odoo/commit/ccc98e01691b88c2da9c484c0e6de235b9d18548
Tracking purpose: it's a bit of a copy of the method below, which also has the `sudo()`. Note that there were no way to share any code between those methods

Reply:
Indeed indeed, forgot about that. Plus in the end, this only returns a bunch of ids, even a wrong sudo would be low risk I guess? Just trying to be extra careful when I see a `sudo`: should I care that you are suddenly exposing the "id" field of all "website.page" to all portal users? :shrug:

Reply:
Probably not that big of a deal: published page ids and name can already be found through sitemap & going to the page and some controllers like get_suggested_links, and for unpublished, it would just be an ID.

But there is no reason to not protect this public method with a check role on restricted editor, I'll add that and test it

Reply:
Added the access protection, kept it below what it should be, see comment, since it's only IDs, seems fine to let restricted editor access it for possible custo?